tools/fluent/basic_syntax.md

```diff
@@ -0,0 +1,131 @@
+# Introduction to Fluent syntax
+
+<!-- toc -->
+
+## Basic syntax
+
+A *message* represents the basic unit of translation in Fluent. This is the simplest form a message could take:
+
+```PROPERTIES
+welcome-msg = Hello, World!
+```
+
+`welcome-msg` is the message identifier (or message ID), and `Hello, World!` is the **value** of this message.
+
+A message can contain **external arguments**, for example:
+
+```PROPERTIES
+welcome-msg = Welcome { $user }
+```
+
+`$user` is the name of the argument, and should never be translated. The fragment included between curly braces is called a **placeable**, and can be moved within the text. For example, the string above would be translated in Italian as follows, leaving the placeable unchanged:
+
+```PROPERTIES
+welcome-msg = Benvenuto { $user }
+```
+
+Messages can also reference other messages. In this case, the placeable won’t include the `$` sign, simply the message identifier:
+
+```PROPERTIES
+menu-settings = Settings
+help-menu-settings = Click “{ menu-settings }” to save the file.
+```
+
+There is a special category of messages, called **private messages**:
+* Private messages have identifiers starting with a dash, e.g. `-brand-short-name`.
+* They can’t be called directly in the source code, but only referenced within other messages.
+
+For example:
+
+```PROPERTIES
+-brand-short-name = Firefox
+close-msg = Close { -brand-short-name }
+```
+
+## Attributes
+
+For a single HTML object there might be multiple messages to localize: for example, a button could have a `label` and an `accesskey`, a link could have a text but also a `tooltip`. Such objects can be localized defining multiple **attributes** within a single message. For example, in the case of a button:
+
+```PROPERTIES
+login-button =
+    .label = Login
+    .accesskey = L
+```
+
+The message `login-button` doesn’t have a *value*, but has 2 *attributes* defined: `label` and `accesskey`. There are a few things to highlight in this FTL fragment:
+* Attribute definitions must be indented and start with a period.
+* There is an equal sign following the message identifier, even if there is no value (i.e. the value is *Null*).
```

Comment:
Maybe add:

> All attributes of public messages found in the `en-US` source must be present in the localization. This will be enforced by tools like `compare-locales`.